src/routes/namespaces/[namespace]/workflows/[workflow]/[run]/stack-trace.svelte

```diff
@@ -0,0 +1,60 @@
+<script context="module" lang="ts">
+  import type { LoadInput } from '@sveltejs/kit';
+  import type { WorkflowExecution } from '$lib/models/workflow-execution';
+
+  export async function load({ stuff }: LoadInput) {
+    const { workflow } = stuff as {
+      workflow: WorkflowExecution;
+    };
+
+    return {
+      props: {
+        workflow,
+      },
+    };
+  }
+</script>
+
+<script lang="ts">
+  import CodeBlock from '$lib/components/code-block.svelte';
+  import Icon from 'svelte-fa';
+  import { namespace } from '$lib/stores/namespace';
+  import { faSyncAlt } from '@fortawesome/free-solid-svg-icons';
+  import Button from '$lib/components/button.svelte';
+  import { getWorkflowStackTrace } from '$lib/services/query-service';
+  import EmptyState from '$lib/components/empty-state.svelte';
+
+  export let workflow: WorkflowExecution;
+
+  let currentdate = new Date();
+  $: datetime = currentdate.toLocaleTimeString();
+  $: data = getWorkflowStackTrace({ workflow, namespace: $namespace });
+
+  const refreshStackTrace = () => {
+    data = getWorkflowStackTrace({ workflow, namespace: $namespace });
+    currentdate = new Date();
+  };
+</script>
+
+<section>
+  {#if String(workflow.status) === 'Running'}
+    <div class="flex items-center">
+      <Button on:click={refreshStackTrace}>
+        <Icon icon={faSyncAlt} scale={0.8} class="block w-full h-full" />
+        Refresh</Button
+      >
+      <p class="ml-9">Stack Trace at {datetime}</p>
+    </div>
+    {#await data}
+      <div>loading</div>
+    {:then { queryResult }}
+      <div class="flex items-start h-full">
+        <CodeBlock
+          content={window.atob(String(queryResult.payloads[0].data))}
```

Comment:
typscript issue here

Reply:
You're telling it that this is a Promise that resolves to null [here](https://github.com/temporalio/ui/pull/187/files#diff-ee9488af68819b334c1e89c9977f90ab4e9f7ce9a3ac2c9498ba02c4ac77b7edR12). If you swap out `null` for the data structure's type, you should be good.

Reply:
For example: https://github.com/temporalio/ui/blob/main/src/lib/services/workflow-service.ts#L106